### PR TITLE
New option to define indentation

### LINKED ISSUE #27: Formatted output with indents?
Hello. Currently object-inspect always represents object as single line.

`{ a: 1, b: [ 3, 4, undefined, null ], c: undefined, d: null, e: { regex: /^x/i, buf: <Buffer 61 62 63>, holes: [ 'a', 'b', , , 'e', , 'g' ] }, now: Thu Nov 14 2019 20:59:42 GMT+1000 (Vladivostok Standard Time), self: [Circular] }`

How about adding the ability to produce formatted output?

```js
{
  a: 1,
  b: [ 3, 4, undefined, null ],
  c: undefined,
  d: null,
  e: {
    regex: /^x/i,
    buf: <Buffer 61 62 63>,
    holes: [ 'a', 'b', <2 empty items>, 'e', <1 empty item>, 'g' ]
  },
  now: 2019-11-14T10:59:42.159Z,
  self: [Circular]
}
```

### PR DESCRIPTION
**Pull request by mkls**
I had a use case where I wanted to see the output of object-inspect indented, 

Current output for `inspect({a: 1, b: 2})`:
```
{ a: 1, b: 2 }
```

New option would make this possible: `inspect({a: 1, b: 2}, {indent: '  '})`:
```
{
  a: 1,
  b: 2
}
```

Added a few test cases that show other cases about how it would work.

If you guys think this would be something useful to include then I'm happy to work on it a bit more and clean it up if needed. Opened this PR is just to test the idea and get some feedback.

Fixes #27.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,6 +50,12 @@ module.exports = function inspect_ (obj, opts, depth, seen) {
         return inspect_(value, opts, depth + 1, seen);
     }
 
+    function indentedJoin(xs) {
+        var previousDepthIndent = Array(depth + 1).join(opts.indent)
```
Comment (ljharb) [inline index.js]: please always include semicolons.

File: index.js [hunk 2]
```diff
@@ -50,6 +50,12 @@ module.exports = function inspect_ (obj, opts, depth, seen) {
         return inspect_(value, opts, depth + 1, seen);
     }
 
+    function indentedJoin(xs) {
```
Comment (ljharb) [inline index.js]: it seems like this function should take `opts.indent` as an argument, so it doesn't need to be recreated every time?

**Comment by mkls** [comment]
I'm also not a 100% sure this is a needed feature, but I found myself missing it. I figured I'll put in a bit of extra effort to open up a PR after I implemented it, just to see if this is needed or not.

I'm not familiar with all the use cases of this module, so I'll gracefully accept if you think this is not something you want to include.

**Comment by deadlyicon** [comment]
@mkls thank you for this! Pease merge! :D

**Pull request merged**